common/lib/xmodule/xmodule/modulestore/tests/test_publish.py

```diff
@@ -139,3 +159,208 @@ def test_publish_draft_delete(self):
+                        test_modulestore,
+                        test_content,
+                        self.course_key,
+                        export_dir,
+                        self.EXPORTED_COURSE_BEFORE_DIR_NAME,
+                    )
+                    # Verify the export OLX format.
+                    self.check_olx_format(export_dir, self.EXPORTED_COURSE_BEFORE_DIR_NAME, self.olx_loc_and_fmt_before)
+
+                    # Get the specified test item.
+                    with test_modulestore.branch_setting(ModuleStoreEnum.Branch.draft_preferred):
+                        test_item = test_modulestore.get_item(self.course_key.make_usage_key(block_type=self.block_type, block_id=self.block_id))
+
+                    # Perform an elemental operation on a particular item.
+                    assert_true(hasattr(test_modulestore, self.operation))
+                    elemental_op = getattr(test_modulestore, self.operation)
+                    op_params = []
+                    for p in self.operation_params:
+                        if isinstance(p, basestring):
+                            if p == '@location':
+                                op_params.append(test_item.location)
+                            elif p == '@user_id':
+                                op_params.append(self.user_id)
+                            else:
+                                op_params.append(p)
+                        else:
+                            op_params.append(p)
+                    elemental_op(*op_params)
+
+                    # Export the course again.
+                    export_course_to_xml(
+                        test_modulestore,
+                        test_content,
+                        self.course_key,
+                        export_dir,
+                        self.EXPORTED_COURSE_AFTER_DIR_NAME,
+                    )
+                    # Verify the export OLX format.
+                    self.check_olx_format(export_dir, self.EXPORTED_COURSE_AFTER_DIR_NAME, self.olx_loc_and_fmt_after)
+
+
+class TestPublishSingleUnit(UniversalTestProcedure):
+    __test__ = False
+    block_type = 'html'
+    block_id = 'unit_0'
+    operation = 'publish'
+    operation_params = ('@location', '@user_id')
+    olx_loc_and_fmt_before = (
+        # Used to check the export before the operation takes place.
+        # List of:
+        #   - path to OLX to check
+        #   - matching OLX
+        (
+            ('drafts', 'html', 'unit_0.xml'),
+            '''
+            <html filename="html_unit"/>
+            '''
+        ),
+        (
+            ('html', 'unit_0.xml'),
+            None,
+        ),
+    )
+    olx_loc_and_fmt_after = (
+        (
+            ('drafts', 'html', 'unit_0.xml'),
+            None, # None means the file/section shouldn't exist.
+        ),
+        (
+            ('html', 'unit_0.xml'),
+            '''
+            <html filename="html_unit"/>
+            '''
+        ),
+    )
```

Comment:
Basically, try and get the individual tests such that they look like straightforward unit tests, even if they're doing a lot of work under the hood.